Sources/ReleaseServer/ReleaseServer.WebApi/Repositories/ReleaseArtifactRepository.cs

```diff
@@ -200,6 +199,53 @@ public List<string> GetPlatforms(string productName, string version)
 
             return platforms.OrderBy(p => p).ToList();
         }
+        
+        public BackupInformationModel RunBackup()
+        {
+            var timeStamp = DateTime.UtcNow.ToString("yyyy-MM-dd-HH-mm-ss");
+            
+            string backupFileName = "backup_" + timeStamp + ".zip";
+            string backupArchiveFileName = Path.Combine(BackupRoot, backupFileName); 
+            
+            //Clear the backup folder first
+            DeleteFilesInPath(BackupRoot);
```

Comment:
This method will throw `System.IO.DirectoryNotFoundException` if the directory under that path does not exist.

Please create a `DirectoryInfo` from the path string here, check if it exists and then delete the contained files.

It would be even more elegant to implement an extension method `DeleteFiles()` for `DirectoryInfo` that performs the deletion of the contained files. This method should check if the directory exists before trying to delete the files in order to avoid throwing an exception.

```suggestion
            var backupDirectoryInfo = new DirectoryInfo(BackupRoot);
            if (backupDirectoryInfo.Exists()) {
                backupDirectoryInfo.DeleteFiles();  // extension method DeleteFiles() has to be implemented
            }
            else {
                backupDirectoryInfo.Create();
            }
```

Reply:
Realized.
See:
https://github.com/Traeger-GmbH/release-server/blob/feature/implement-backup-mechanism/Sources/ReleaseServer/ReleaseServer.WebApi/Repositories/ReleaseArtifactRepository.cs#L215-L225